### PR TITLE
feat: display warning for failure to send a message to an offline back-end [FS-1989]

### PR DESCRIPTION
**Pull request by V-Gira**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/FS-1989" title="FS-1989" target="_blank">BugFS-1989</a>  [Web] Learn more link is missing from "Message could not be sent, as the backend of domain could not be reached."
  </td></table>
  <br />
 

### Issues

- A specific error message for a complete failure to send when the recipient b-e is offline wasn't implemented
image

- A link to the help center is missing in a couple places

### CODE REVIEW

File: src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx [hunk 1]
```diff
@@ -186,9 +186,9 @@ const ContentMessageComponent: React.FC<ContentMessageProps> = ({
           <PartialFailureToSendWarning failedToSend={failedToSend} knownUsers={conversation.allUserEntities()} />
         )}
 
-        {status === StatusType.FAILED && (
+        {(status === StatusType.FAILED || status === StatusType.FEDERATION_ERROR) && (
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx]:
```suggestion
        {[StatusType.FAILED, StatusType.FEDERATION_ERROR].includes(status) && (
```

File: src/script/components/MessagesList/Message/ContentMessage/Warnings/CompleteFailureToSend/CompleteFailureToSend.tsx [hunk 2]
```diff
@@ -17,27 +17,47 @@
...
 
 type Props = {
-  isTextAsset: boolean;
   onRetry: () => void;
+  unreachableDomain?: string;
 };
 
-export const CompleteFailureToSendWarning = ({isTextAsset, onRetry}: Props) => {
+export const CompleteFailureToSendWarning = ({onRetry, unreachableDomain}: Props) => {
+  const config = Config.getConfig();
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/Warnings/CompleteFailureToSend/CompleteFailureToSend.tsx]: Hmm.. I think You can move it outside component :) to not rerender it

File: src/script/components/MessagesList/Message/ContentMessage/Warnings/PartialFailureToSend/PartialFailureToSend.tsx [hunk 3]
```diff
@@ -76,6 +77,8 @@ export const PartialFailureToSendWarning = ({failedToSend, knownUsers}: Props) =
   const [isOpen, setIsOpen] = useState(false);
   const {queued = {}, failed = []} = failedToSend;
 
+  const config = Config.getConfig();
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/ContentMessage/Warnings/PartialFailureToSend/PartialFailureToSend.tsx]: Try to move it before component :)

File: src/script/conversation/MessageRepository.ts [hunk 4]
```diff
@@ -807,7 +808,7 @@ export class MessageRepository {
       }
       return result;
     } catch (error) {
-      await this.updateMessageAsFailed(conversation, payload.messageId);
+      await this.updateMessageAsFailed(conversation, payload.messageId, error as BackendError);
```
Comment (PatrykBuniX) [inline src/script/conversation/MessageRepository.ts]: Can we do an if statement here instead of assertion? There's `isBackendError(error)` util 😄

**Review by przemvs — APPROVED** [review]

**Review by PatrykBuniX — APPROVED** [review]

File: src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx [hunk 5]
```diff
@@ -186,9 +186,9 @@ const ContentMessageComponent: React.FC<ContentMessageProps> = ({
           <PartialFailureToSendWarning failedToSend={failedToSend} knownUsers={conversation.allUserEntities()} />
         )}
 
-        {status === StatusType.FAILED && (
+        {[StatusType.FAILED, StatusType.FEDERATION_ERROR].includes(status) && (
           <CompleteFailureToSendWarning
-            isTextAsset={message.getFirstAsset().isText()}
+            {...(status === StatusType.FEDERATION_ERROR && {unreachableDomain: conversation.domain})}
             onRetry={() => onRetry(message)}
```
Comment (arjita-mitra) [inline src/script/components/MessagesList/Message/ContentMessage/ContentMessage.tsx:197]: need to send isMessageFocused prop to ensure correct focus order

File: src/script/components/MessagesList/Message/ContentMessage/Warnings/PartialFailureToSend/PartialFailureToSend.tsx [hunk 6]
```diff
@@ -148,7 +151,16 @@ export const PartialFailureToSendWarning = ({failedToSend, knownUsers}: Props) =
                   )}
                   {unreachableUsers.length === 1
                     ? ` ${t('messageFailedToSendWillNotReceiveSingular')}`
-                    : ` ${t('messageFailedToSendWillNotReceivePlural')}`}
+                    : ` ${t('messageFailedToSendWillNotReceivePlural')}`}{' '}
+                  <Link
+                    targetBlank
+                    variant={LinkVariant.PRIMARY}
```
Comment (arjita-mitra) [inline src/script/components/MessagesList/Message/ContentMessage/Warnings/PartialFailureToSend/PartialFailureToSend.tsx:162]: need to pass messageFocusedTabIndex

File: src/script/components/MessagesList/Message/ContentMessage/Warnings/CompleteFailureToSend/CompleteFailureToSend.tsx [hunk 7]
```diff
@@ -17,27 +17,48 @@
...
+          >
+            {t('offlineBackendLearnMore')}
+          </Link>
+        </p>
+      ) : (
+        <p>{t('messageCouldNotBeSentConnectivityIssues')}</p>
+      )}
+      <div css={{display: 'flex'}}>
+        <Button type="button" variant={ButtonVariant.TERTIARY} onClick={onRetry}>
+          {t('messageCouldNotBeSentRetry')}
```
Comment (arjita-mitra) [inline src/script/components/MessagesList/Message/ContentMessage/Warnings/CompleteFailureToSend/CompleteFailureToSend.tsx:64]: need to pass messageFocusedTabIndex

File: src/script/components/MessagesList/Message/ContentMessage/Warnings/CompleteFailureToSend/CompleteFailureToSend.tsx [hunk 8]
```diff
@@ -17,27 +17,48 @@
...
+        <p>
+          <span
+            css={warning}
+            dangerouslySetInnerHTML={{
+              __html: t('messageCouldNotBeSentBackEndOffline', {domain: unreachableDomain}),
+            }}
+          />{' '}
+          <Link
+            targetBlank
+            variant={LinkVariant.PRIMARY}
```
Comment (arjita-mitra) [inline src/script/components/MessagesList/Message/ContentMessage/Warnings/CompleteFailureToSend/CompleteFailureToSend.tsx:51]: need to pass messageFocusedTabIndex

**Review by arjita-mitra — CHANGES_REQUESTED** [review]

**Review by arjita-mitra — APPROVED** [review]

**Pull request merged**